test/conformance/enqueue/urEnqueueMemBufferMap.cpp

```diff
@@ -2,12 +2,52 @@
 // Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
 // See LICENSE.TXT
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "helpers.h"
 #include <uur/fixtures.h>
 
-using urEnqueueMemBufferMapTest = uur::urMemBufferQueueTest;
-UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urEnqueueMemBufferMapTest);
-
-TEST_P(urEnqueueMemBufferMapTest, SuccessRead) {
+using urEnqueueMemBufferMapTestWithParam =
+    uur::urMemBufferQueueTestWithParam<uur::mem_buffer_test_parameters_t>;
+
+static std::vector<uur::mem_buffer_test_parameters_t> test_parameters{
+    {8, UR_MEM_FLAG_READ_WRITE},
+    {16, UR_MEM_FLAG_READ_WRITE},
+    {32, UR_MEM_FLAG_READ_WRITE},
+    {64, UR_MEM_FLAG_READ_WRITE},
+    {128, UR_MEM_FLAG_READ_WRITE},
+    {256, UR_MEM_FLAG_READ_WRITE},
+    {512, UR_MEM_FLAG_READ_WRITE},
+    {1024, UR_MEM_FLAG_READ_WRITE},
+    {8, UR_MEM_FLAG_WRITE_ONLY},
```

Comment:
The idea here was so a range of buffer sizes could be tested but something I didn't initially consider was just how many tests would be spun up from the amount of variations here. Before this change this particular test suite had 11 tests, but this change would increase that up to 384 which would mean a lot more entries to the match files unless using a wildcard to include them all in one line. Would this be fine or better to choose less buffer sizes?

Reply:
I was just asking @callumfare about this. I didn't know that the smaller sizes would probably be rounded up so didn't realise many of these entries are redundant. I will change this to just be a couple different larger sizes instead, so it's still similar to what the E2E tests were doing.